wd_helper.au3

```diff
@@ -636,6 +636,173 @@ Func _WD_FrameLeave($sSession)
 	Return SetError(__WD_Error($sFuncName, $_WD_ERROR_Success), 0, $sValue)
 EndFunc   ;==>_WD_FrameLeave
 
+; #FUNCTION# ====================================================================================================================
+; Name ..........: _WD_FrameList
+; Description ...: get frame list from current webdriver context/tab
+; Syntax ........: _WD_FrameList($sSession[, $bReturnAsArray = True[, $sFilter = ''[, $bReturnHTML = False]]])
+; Parameters ....: $sSession            - Session ID from _WD_CreateSession
+;                  $bReturnAsArray      - By default on True function will return array, other wise string
+;                  $sFilter             - RegExp pattern to check if desired content exist in frame document HTML source, if used it returns only frames with diesired content
+;                  $bReturnHTML         - On true function will return frame document HTML source
+; Return values .: Success - array or string
+;                  Failure - "" (empty string) and sets @error to one of the following values:
+;                  - $_WD_ERROR_GeneralError
+;                  - $_WD_ERROR_Exception
+;                  - $_WD_ERROR_InvalidExpression
+;                  - $_WD_ERROR_NotFound
+; Author ........: mLipok
+; Modified ......:
+; Remarks .......:
+; Related .......: __WD_FrameList_Internal, _WD_FrameEnter, _WD_GetFrameCount
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func _WD_FrameList($sSession, $bReturnAsArray = True, $sFilter = '', $bReturnHTML = False)
+	Local Const $sFuncName = "_WD_FrameList"
+	Local Const $sParameters = 'Parameters:    ReturnAsArray=' & $bReturnAsArray & '   Filter=' & $sFilter & '   ReturnHTML=' & $bReturnHTML
+	Local $a_Result[0][6], $sStartLocation = '', $sMessage = ''
+	Local $vResult = '', $iErr = $_WD_ERROR_Success, $iExt = 0
+	Local $sStartBody_ElementID = _WD_ExecuteScript($sSession, "return window.document.body;", Default, Default, $_WD_JSON_Element)
+	If @error Then
+		$iErr = $_WD_ERROR_GeneralError
+	Else
+		$vResult = __WD_FrameList_Internal($sSession, 'null', '', $sFilter, $bReturnHTML)
```

Comment:
`$sFilter` should not be maintained internally because then we loose `$sStartLocation` setting feature and for this reason

https://github.com/Danp2/au3WebDriver/blob/74f6fb0c88c2cfd426b9e10b6604731e77b77c03/wd_helper.au3#L697-L700

will be omnited

Reply:
Are you saying that you no longer intend to restore the browsing context before exiting?

Reply:
No. I say that even if `$sFilter` is used it should back and for this reason I need to refactor this by move `$sFilter` functionality out of `__WD_FrameList_Internal` to `_WD_FrameList`.

btw.
WIP

Reply:
fixed.
Check current version